### ISSUE TITLE
Outputs: -M with any options, should keep things consistent & show progress.

### ISSUE THREAD

**Issue by tekdj7** [prowler-cloud/prowler#614]
**prowler -M text** shows banner with version, date,  and color code (keeping consistency). It also shows pass/fail/info as it processes the different resources/regions for each check (shows progress). This is perfect.

**prowler -M html** missing banner with version, date, and color code. (not consistent). Also missing the pass/fail/info as it processes the different resources/regions for each check (progress not shown as granular as before)

I do like the fact that **prowler -M text,html** shows everything perfectly, and still saves out HTML report.  

**prowler -M text,html,csv** results shown in the screen are inconsistent, not pretty to view as progress.  It should instead always show the same consistent/progress view as **prowler -M text**.  Instead, its showing the output as its being written to a file based on the last option in the -M option.  So in the case of csv, I see csv lines in console output.

**prowler -M text,html,json** same feedback as **csv**, except since its json, its showing json lines in console output.

**prowler -M text,html,json** same feedback as **json-asff**, except since its json-asff, its showing json-ASFF lines in console output.

**Comment by stuart-shelton-contino** [comment]
Further to this:

```
prowler -M text,html,json,json-asff,junit-xml,csv -b -n -s
```

... outputs:

```

 Writing JUnit XML reports to /prowler/junit-reports 

Generating "," delimited report on stdout for profile {xxx}, account {yyy}
```

... and then proceeds to output CSV data to the console.

Files written (as, e.g., `prowler-output--20201009135349.csv` - missing the AWS account ID when `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` are taken from the environment) are:

```
prowler-output--20201009135349.asff-json
prowler-output--20201009135349.csv
prowler-output--20201009135349.html
prowler-output--20201009135349.json
```


... but there's no `/prowler/junit-reports`.

So it looks as if `csv` overrides `text`, even though CSV data is also being written to a file whilst `text` output is lost, and one of the additional output formats is preventing `junit-xml` from producing output.

At the very least, the documentation should state the the ordering of output formats is significant.

**Comment by stuart-shelton-contino** [comment]
... with:

```
prowler -M html,json,json-asff,csv,junit-xml,text -b -n -s
```

The console output is a mix of CSV and XML 😔

(You'll note that the only thing that's moved is `text`, from the beginning of the list to the end.  But the output has changed from CSV to CSV/XML intermixed - so the output isn't even consistent.

It feels as if the issue here is that several settings are being conflated, and that's causing the confusion - would it be better to have separate options for console output (accepting one format only, or possibly one or more of format, banner, and/or summary?) and file creation?

e.g. `prowler --screen-output=ansi,banner,summary --file-output=html,json,json-asff,csv,junit-xml,text,ansi`

... noting that I've changed `test` to `ansi` to better explain it's function, which also allows for `text` to represent plain-text.  It feels as if both plain-text and coloured output should be possible... perhaps the extension for ANSI-formatted text should be `.ans` (... which appears to be a semi-standard)?

You could even allow output streams to be named in this way:

`prowler --screen-output=banner,text,summary --file-output=/mnt/prowler.json --file-output=/mnt/prowler.txt,ansi`

... where `--file-output` takes one or two comma-separated arguments: if one, if just end in a recognised file extension which determines the format of the output.  If two, then the first represents the output filename (which needn't have a recognised extension) and the second determines the format of the content.

**Comment by stuart-shelton-contino** [comment]
However, `prowler -M html,json,json-asff,csv,junit-xml,text -b -n -s` *does* produce `junit-reports`, which didn't happen with `text` at the start of the list!

**Comment by toniblyx** [comment]
Hi @stuart-shelton-contino you are right, the console output when using multiple formats is not fine tuned or properly tested to have a consistent behavior. What I think it would be easier to implement is to leave the default colored console output (no mater the chosen output format) and anything else to a file inside the output directory. What do you think? I'm going to make some tests.

**Comment by toniblyx** [comment]
hey @stuart-shelton-contino I have it almost done with that but I'm panicking about pushing it to master because I made a lot of changes in whoami and outputs 🤣  it needs lot of testing. Can you help with that?

**Comment by stuart-shelton-contino** [comment]
Apologies, I've only just seen this message (my GitHub alert emails either didn't fire or got lost)!

If I'm not too late I'd be happy to test ☺️

**Comment by toniblyx** [comment]
I created a branch here https://github.com/toniblyx/prowler/tree/new-output-behavior with some changes, that hides all outputs.

**Comment by toniblyx** [comment]
Most of the asked features here were implemented for 2.5. Outputs will keep improving for better consistency but now, regardless the output asked there is a default output in the console and files are generated in the background.

Show progress itself will be considered for v3 once the new checks architecture is in place (grouping checks by AWS service instead of based on CIS)
